### PR TITLE
#287: Buildo.io CI migration to Concourse (closes #287)

### LINKED ISSUE #287: Buildo.io CI migration to Concourse
Buildo.io seems to be still on drone CI.

### PR DESCRIPTION
**Pull request by ecamellini**
Closes #287

### tests performed

> A Test Plan is used to show what you tested to make sure your code works fine. You should write here all that you did to test, and provide some results of your testing.

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

### tests not performed (domain coverage)

> At times not everything can be tested, and writing what hasn't been tested is just as important as writing what has been tested.

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

**Comment by ecamellini** [comment]
@gabro I started this in the free time, not sure of the deploy part (I think I need to add some Cloudflare related stuff?)

File: ci/build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+set -e
+
+cd buildo.io
```
Comment (gabro) [inline ci/build.sh]: you can use `dir` in `build.yml` to avoid the `cd` step

File: ci/build.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - name: buildo.io
+
+caches:
+  - path: buildo.io/web/node_modules
+
+outputs:
+  - name: build
+
+run:
+  path: buildo.io/ci/build.sh
```
Comment (gabro) [inline ci/build.yml]:
```suggestion
  path: buildo.io/ci/build.sh
  dir: buildo.io
```

File: ci/build.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: node
+    tag: "8"
```
Comment (gabro) [inline ci/build.yml]: can you try with `10-alpine` instead? If it doesn't work, you can at least use `10-slim`

File: ci/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+jobs:
+  - name: pr-build
+    plan:
+      - get: buildo.io
+        resource: pr
+        trigger: true
+        version: every
+      - get: node
+        params:
+          save: true
```
Comment (gabro) [inline ci/pipeline.yml]: I don't think this resource is used.

File: ci/pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            context: concourse
+
+  - name: deploy
+    plan:
+      - get: buildo.io
+        resource: master
+        trigger: true
+      - get: node
+        params:
+          save: true
```
Comment (gabro) [inline ci/pipeline.yml]: same, I don't think the `node` resource is ever used

File: ci/preview.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+platform: linux
+
+image_resource: ## TODO replace with an image with just JQ such as stedolan/jq
+  type: docker-image
+  source:
+    repository: node
+    tag: "8"
```
Comment (gabro) [inline ci/preview.yml]: same as above.

File: ci/preview.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+platform: linux
+
+image_resource: ## TODO replace with an image with just JQ such as stedolan/jq
```
Comment (gabro) [inline ci/preview.yml]: I think it's fine to use node and install jq every time, it's lightweight and it allows us to use the official node image.

File: .drone.yml [hunk 8]
```diff
@@ -1,51 +0,0 @@
...
-        exclude: master
-
-  deploy-prod:
-    image: plugins/s3-sync:1
-    bucket: www.buildo.io
-    secrets: [ aws_access_key_id, aws_secret_access_key ]
-    region: eu-central-1
-    source: build
-    target: /
-    cloudfront_distribution: E370OAVTAU5FQ4
```
Comment (ecamellini) [inline .drone.yml]: @gabro do you know if this cloudfront param and the svg-related one below need to be added also to the new script?

File: ci/build.yml (hunk 3 above)
Comment (ecamellini) [reply ci/build.yml]: I get the following error both with 10-alpine and 10-slim 
image

File: ci/build.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  - name: buildo.io
+
+caches:
+  - path: buildo.io/web/node_modules
+
+outputs:
+  - name: build
+
+run:
+  path: buildo.io/ci/build.sh
```
Comment (gabro) [inline ci/build.yml]: my mistake: this should be

```yaml
path: ci/build.sh
dir: buildo.io
```

This is the reason for the error you're getting (the docker image is irrelevant)

Comment (ecamellini) [reply ci/build.yml]: Makes sense, reading the error would have helped, sorry

File: ci/build.yml (hunk 3 above)
Comment (ecamellini) [reply ci/build.yml]: Still didn't manage to make it work, getting weird node errors, maybe I can debug them better after I make everything else work with node `8`.

**Comment by ecamellini** [comment]
Preview works right now:

image

File: ci/build.yml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: node
+    tag: 8
```
[the remark this answers is not in the document]
Comment (ecamellini) [reply ci/build.yml]: Yep, I tried both alpine and slim, both 8 and 10, but get some node errors, I could debug a little bit more

**Pull request merged**